From 1a70f76671794c17cc2f5038310cd2ef81c4464f Mon Sep 17 00:00:00 2001
From: Hieu Huynh <hieuh@os.amperecomputing.com>
Date: Tue, 30 Nov 2021 04:55:40 +0000
Subject: [PATCH] Add d-bus OEM for boot progress

This commit adds the BootProgressOEM for the OEM-specific LastState and
the OEM enumerations for the ProgressStages as defined in the
DEN0069C_SBMR_1.1 document.

Signed-off-by: Hieu Huynh <hieuh@os.amperecomputing.com>
---
 .../openbmc_project/State/Boot/Progress.interface.yaml   | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/yaml/xyz/openbmc_project/State/Boot/Progress.interface.yaml b/yaml/xyz/openbmc_project/State/Boot/Progress.interface.yaml
index 07f2d82..1b9415f 100644
--- a/yaml/xyz/openbmc_project/State/Boot/Progress.interface.yaml
+++ b/yaml/xyz/openbmc_project/State/Boot/Progress.interface.yaml
@@ -18,6 +18,12 @@ properties:
           This can be compared with the current BootProgress value
           to know how long the boot has been on the current boot step.
 
+    - name: BootProgressOem
+      type: string
+      description: >
+          This property shall represent the OEM-specific LastState of
+          the BootProgress.
+
 enumerations:
     - name: ProgressStages
       description: >
@@ -56,3 +62,6 @@ enumerations:
           - name: MotherboardInit
             description: >
                 Motherboard/Basebord Initialization.
+          - name: OEM
+            description: >
+                OEM-defined boot progress state.
-- 
2.17.1

